usaspending_api/etl/transaction_loaders/tests/integration/test_fpds_loader.py

```diff
@@ -48,36 +55,59 @@ def test_load_ids_from_broker(self):
-        # assert 201 in [_.transaction.id for _ in usaspending_transactions]
-        # assert 301 in [_.transaction.id for _ in usaspending_transactions]
-
-        for obj in TransactionFPDS.objects.all():
-            self.logger.info("Found FPDS Transaction with id = {}".format(obj.transaction.id))
-
-        # TODO:Make asserts on the data
-        # Compare data is as expected
-        # assert load_objects_pre_transaction["award"]["transaction_unique_id"] == str(dummy_broker_ids[0])
-        # assert load_objects_pre_transaction["transaction_normalized"]["transaction_unique_id"] == str(dummy_broker_ids[0])
-        # assert load_objects_pre_transaction["transaction_normalized"]["award_id"] == final_award_id
-        # assert load_objects_pre_transaction["transaction_normalized"]["funding_agency_id"] == 1
-        # assert load_objects_pre_transaction["transaction_normalized"]["awarding_agency_id"] == 1
-        # assert 2001 <= load_objects_pre_transaction["transaction_normalized"]["fiscal_year"] <= 2019
+
+        # Run core logic to be tested
+        call_command("load_fpds_transactions", "--ids", *broker_id_list)
+
+        # Assert results
+
+        # Lineage should trace back to the broker records
+        usaspending_transactions = TransactionFPDS.objects.all()
+        assert len(usaspending_transactions) == 3
+        tx_fpds_broker_ref_ids = [_.detached_award_procurement_id for _ in usaspending_transactions]
+        assert 101 in tx_fpds_broker_ref_ids
+        assert 201 in tx_fpds_broker_ref_ids
+        assert 301 in tx_fpds_broker_ref_ids
+        tx_fpds_broker_ref_id_strings = [_.detached_award_proc_unique for _ in usaspending_transactions]
+        assert "101" in tx_fpds_broker_ref_id_strings
+        assert "201" in tx_fpds_broker_ref_id_strings
+        assert "301" in tx_fpds_broker_ref_id_strings
+        tx_norm_broker_ref_id_strings = [_.transaction.transaction_unique_id for _ in usaspending_transactions]
+        assert "101" in tx_norm_broker_ref_id_strings
+        assert "201" in tx_norm_broker_ref_id_strings
+        assert "301" in tx_norm_broker_ref_id_strings
+
+        # All 3 should be under the same award
+        # 1 award created
+        usaspending_awards = Award.objects.all()
+        assert len(usaspending_awards) == 1
+        new_award = usaspending_awards[0]
+
+        # All transactions reference that award and its id
+        tx_norm_awd = [_.transaction.award for _ in usaspending_transactions]
+        assert usaspending_awards[0] == tx_norm_awd[0] == tx_norm_awd[1] == tx_norm_awd[2]
+        tx_norm_awd_ids = [_.transaction.award_id for _ in usaspending_transactions]
+        assert new_award.id == tx_norm_awd_ids[0] == tx_norm_awd_ids[1] == tx_norm_awd_ids[2]
+
+        # And the single award created should refer to the first transaction processed that spawned the award (101)s
+        # TODO: Check if this is true??
+        assert new_award.transaction_unique_id == "101"
```

Comment:
@tony-sappe @kbard @BrianZito - can one of you confirm this outcome and assertion is correct?

Reply:
Looks right to me.  Is it failing?

In `test_fpds_loader`:
```
        broker_id_list = [101, 201, 301]
        broker_objects = _assemble_detached_award_procurement_records(broker_id_list)
```
Then in `_assemble_detached_award_procurement_records` in `test_fpds_loader`:
```
    for id in id_list:
        dummy_row = _assemble_dummy_broker_data()
        dummy_row["detached_award_procurement_id"] = id
        dummy_row["detached_award_proc_unique"] = str(id)
```
Then in the loader in `field_mappings_fpds`:
```
# broker column name -> usaspending column name
transaction_normalized_nonboolean_columns = {
    "period_of_performance_star": "period_of_performance_start_date",
    ...
    "detached_award_proc_unique": "transaction_unique_id",
}
```
Which is used by `fpds_loader` to map Broker fields to USAS fields.

Reply:
yeah, that looks accurate. The value will be set by the loader when looking up if an award exists by the transaction's generated_unique_award_id ... and if not found, create the award referencing that spawning-transaction-id.

But does anyone know why this field is necessary? What use is it associating an award with the _first_ (and only the first) transaction that came about under this award (which may or may not be the earliest or latest transaction). Do we use this in queries in some way, and for what?

Reason I'm asking is that the logic for creation seems arbitrary/brittle, so if we have important queries using this field, we should be sure they're making the right assumptions about what this field references.

@BrianZito @tony-sappe @kbard

Reply:
Do we care? dev-3464 is to get something stable and faster for FPDS loads, and we already have an epic to remove this table structure entirely. I'd argue the viability of these fields is out of scope at this point.